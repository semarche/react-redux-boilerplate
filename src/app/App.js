import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import debounce from 'lodash/debounce';
import Layout from '../layout';
import LogIn from '../login';
import './style.scss';

const propTypes = {
  setWidth: PropTypes.func.isRequired,
  size: PropTypes.object.isRequired,
};

class App extends Component {
  constructor(props) {
    super(props);
    this.changeWidth = debounce(width => props.setWidth(width), 200);
  }
  componentWillReceiveProps(nextProps) {
    const { size: { width } } = this.props;
    if (nextProps.size.width !== width) {
      this.changeWidth(nextProps.size.width);
    }
  }
  componentDidMount() {
    const { setWidth, size: { width } } = this.props;
    setWidth(width);
  }
  render() {
    return (
      <div className='app'>
        <Switch>
          <Route exact path='/login' component={LogIn} />
          <Route path='/' component={Layout}/>
        </Switch>
      </div>
    )
  }
}

App.propTypes = propTypes;
export default App;
