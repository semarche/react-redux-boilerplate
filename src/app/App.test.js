import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import App from './App';

configure({ adapter: new Adapter() });

describe('App', () => {
  const wrapper = shallow(
    <App
      setWidth={jest.fn()}
      size={ { size: { width: 1200 } } }
    />
  )

  it('should render App component', () => {
    expect(wrapper.find('.app').exists()).toEqual(true);
  });
  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
