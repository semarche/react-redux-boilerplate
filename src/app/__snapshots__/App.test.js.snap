// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="app"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/"
    />
  </Switch>
</div>
`;
